lib/vrl/parser/src/lex.rs

```diff
@@ -1069,7 +1074,7 @@ impl<'input> Lexer<'input> {
 // -----------------------------------------------------------------------------
 
 fn is_ident_start(ch: char) -> bool {
-    matches!(ch, 'a'..='z')
+    matches!(ch, '@' | 'a'..='z')
```

Comment:
I'm surprised this doesn't break anything else. Given that we now have different definitions of identifiers.

But, if all tests are still passing, then we should be good.

Reply:
> I'm surprised this doesn't break anything else. Given that we now have different definitions of identifiers.

The differences are handled [here](https://github.com/timberio/vector/blob/17d0fa0027d3594b825506193acea79c04f29c94/lib/vrl/parser/src/lex.rs#L429). The lexer for identifiers doesn't know what type of identifier it is parsing until it has finished, so here we just have to accept all possible characters and then differentiate at the end..

Reply:
That's partially true, but we also delegate to this function in `is_ident_continue` and `is_query_start`, which are used in other places. But I have nothing that comes to mind that would break because of this, and given that the tests are green, I don't see a reason to change this 👍